### PR TITLE
zsh-fast-syntax-highlighting 1.55 (new formula)

### PR DESCRIPTION
**Pull request by gwerbin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by gwerbin** [comment]
I don't understand the "GitHub repository too new" error.

There was recently somewhat of a kerfuffle where the original maintainer of this (and a few other) projects decided to take down all the source code. This repo is a community fork of a long-established project. See here for info: <https://github.com/zdharma-continuum/I_WANT_TO_HELP>.

File: Formula/zsh-fast-syntax-highlighting.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class ZshFastSyntaxHighlighting < Formula
+  desc "Feature-rich syntax highlighting for Zsh"
+  homepage "https://github.com/zdharma-continuum/fast-syntax-highlighting"
+  url "https://github.com/zdharma-continuum/fast-syntax-highlighting.git",
+      tag:      "v1.55",
+      revision: "5351bd907ea39d9000b7bd60b5bb5b0b1d5c8046"
```
Comment (carlocab) [inline Formula/zsh-fast-syntax-highlighting.rb:6]: Do we really need a Git clone?

Comment (gwerbin) [reply Formula/zsh-fast-syntax-highlighting.rb:6]: No, we don't. I was emulating the `zsh-syntax-highlighting` formula which did the same. We can use the usual Github tarball.

File: Formula/zsh-fast-syntax-highlighting.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  uses_from_macos "zsh" => [:build, :test]
+
+  def install
+    pkgshare.install Dir["*"]
+  end
+
+  def caveats
+    <<~EOS
+      To activate the syntax highlighting, add the following at the end of your .zshrc:
+        source #{HOMEBREW_PREFIX}/share/zsh-fast-syntax-highlighting/fast-syntax-highlighting.plugin.zsh
```
Comment (carlocab) [inline Formula/zsh-fast-syntax-highlighting.rb:19]: If we install this file using `zsh_function.install` then one would only need to add `$HOMEBREW_PREFIX/share/zsh/site-functions` to `FPATH`. If you use Homebrew Zsh you don't even need that (it's already in `FPATH`).

Comment (gwerbin) [reply Formula/zsh-fast-syntax-highlighting.rb:19]: I didn't know we had `zsh_function` 😄 . I will definitely consider using it for other Zsh things. However that isn't the objective here.

This is meant to be `source`d once, in the user's config, upon startup. The user might want to do that manually in zshrc (as shown here), or by a plugin manager such as Antigen, Zinit, etc.

Comment (carlocab) [reply Formula/zsh-fast-syntax-highlighting.rb:18]: AH, right. Misread the `.plugin.zsh`.

File: Formula/zsh-fast-syntax-highlighting.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <<~EOS
+      To activate the syntax highlighting, add the following at the end of your .zshrc:
+        source #{opt_pkgshare}/fast-syntax-highlighting.plugin.zsh
+    EOS
+  end
+
+  test do
+    test_script = \
+      "zsh -c 'source #{pkgshare}/fast-syntax-highlighting.plugin.zsh" \
+      " && echo ${FAST_HIGHLIGHT_STYLES+yes}'"
```
Comment (carlocab) [inline Formula/zsh-fast-syntax-highlighting.rb:25]: Can we make this into a proper script with something like
```ruby
(testpath/"script.zsh").write <<~EOS
  # script contents here
EOS
```
?

**Pull request closed without merging**